metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -220,13 +220,15 @@ private boolean isValidNewLeader(int replica) {
     private void tryElection(PartitionChangeRecord record) {
         ElectionResult electionResult = electLeader();
         if (electionResult.node != partition.leader) {
-            log.debug(
-                "Setting new leader for topicId {}, partition {} to {} using {} election",
-                topicId,
-                partitionId,
-                electionResult.node,
-                electionResult.unclean ? "an unclean" : "a clean"
-            );
+            if (log.isTraceEnabled()) {
+                log.trace(
+                    "Setting new leader for topicId {}, partition {} to {} using {} election",
+                    topicId,
+                    partitionId,
+                    electionResult.node,
+                    electionResult.unclean ? "an unclean" : "a clean"
```

Comment:
I wonder if it would make sense to log unclean elections at a higher level?

Reply:
I don't think it is possible to do so since we don't know the result of the election anywhere else.  Unclean leader election may be enabled, for example, but it may not ever occur.  It is only here that we know.  I added additional logic to ensure that we will always log an unclean election at the WARN level.